src/casm/crystallography/BasicStructure.cc

```diff
@@ -20,6 +20,8 @@ BasicStructure BasicStructure::from_poscar_stream(std::istream &poscar_stream,
                                                   double tol) {
   BasicStructure poscar_structure;
   poscar_structure.read(poscar_stream, tol);
+  poscar_structure.set_unique_names(
+      allowed_molecule_unique_names(poscar_structure));
```

Comment:
@bpuchala , when reading `BasicStructure` from `poscar_stream`, `unique_names` is not getting set anywhere.
`Prim.occ_dofs()` in python uses `this->unique_names()` for getting `occ_dofs` and comes up empty because of this.
Hence I am calling `allowed_molecule_unique_names` here. Let me know if there's a better way of doing it.